lib/active_model/serializer/reflection.rb

```diff
@@ -106,42 +116,44 @@ def value(serializer)
       #
       # @api private
       #
-      def build_association(subject, parent_serializer_options)
-        association_value = value(subject)
+      def build_association(parent_serializer, parent_serializer_options)
+        association_value_proc = proc do
+          value(parent_serializer)
+        end
+        serializer_proc = ->(association_value, reflection_options) {
```

Comment:
hack to allow minimal code changes while lazy-evaluating the association value and its derived serializer